### ISSUE TITLE
Missing call to model.optimizer._clip_gradients(grads)?

### ISSUE THREAD

**Issue by jponf** [tensorflow/tensorflow#45146]
Hi,

I've been having problems training a model and, after reading the keras training loop code, it seems that the call to 
model.optimizer._clip_gradients(grads) introduced in 69da929ad4d5ba605507efa1f52b382a55b6a969 is missing again.

Has it been moved somewhere else?

https://github.com/tensorflow/tensorflow/blob/ab953ab8c5612c67b07e4159fb9eed9edef089ff/tensorflow/python/keras/engine/training_eager_v1.py#L278

**Comment by Saduf2019** [comment]
@jponf 
We see that the issue template has not been filled, could you please do so as it helps us analyse the issue [tf version, steps followed before you ran into this error or stand alone code to reproduce the issue faced]

**Comment by jponf** [comment]
Hi, the behavior was already explained on a previous issue #33929, were it is mentioned that TF 2.2 fixed it. But it seems that it is happening again in TF 2.3.0

* TF version: 2.3.0
* TF Keras version: 2.4.0

I was training a CRNN and tried to apply clipnorm to avoid exploding gradients, but the issue persisted after adding the parameter to the optimizer. Then I just sub-classed tf.keras.Model and modified the `train_step` method to apply `tf.clip_by_global_norm` before calling the optimizer and so far it seems the exploding gradients problem is gone.

I've run the code provided in #33929 (https://colab.research.google.com/drive/16gxaqBuY7NRYZaDqeV26yH7caLChTLY6) and when training with clipvalue=0 the loss changes, which to my understanding it shouldn't happen.

**Comment by Saduf2019** [comment]
@jponf 
I ran the code on tf-nightly and [tf 2.3](https://colab.research.google.com/gist/Saduf2019/ccbeccee471ec883c4deb875538cc173/untitled476.ipynb) do not see the nan values, please refer to this [gist here](https://colab.research.google.com/gist/Saduf2019/77fb086287b1846cf61302ad3a2f3f93/untitled472.ipynb).

**Comment by jponf** [comment]
In my particular case it wasn't NaNs what I found, but the running mean of the batchnorm layers that skyrocketed to values between 1000 and 300000, even though the input was normalized between 0 and 1. 

A complete rewrite of the model in PyTorch (with gradient clipping) did not has this issue, and as I said, a custom `train_step` with `tf.clip_by_global_norm` before applying the gradients didn't show that behavior either. That is what lead me to the issue mentioned in my previous comment.

If my schedule allows me to, I'll try to explore this issue in more detail. 

In the meanwhile, what I found is that line 277, were gradient clipping is applied, in
https://github.com/tensorflow/tensorflow/blob/69da929ad4d5ba605507efa1f52b382a55b6a969/tensorflow/python/keras/engine/training_eager.py
is gone in:
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/keras/engine/training_eager_v1.py

Which may be completely unrelated to my problem but so far is the only thing that I know may explain the behavior that I observed.
